common/src/main/java/org/apache/falcon/metadata/InstanceRelationshipGraphBuilder.java

```diff
@@ -76,8 +76,8 @@ public Vertex addProcessInstance(WorkflowExecutionContext context) throws Falcon
         String processInstanceName = getProcessInstanceName(context);
         LOG.info("Adding process instance: {}", processInstanceName);
 
-        Vertex processInstance = addVertex(processInstanceName,
-                RelationshipType.PROCESS_INSTANCE, context.getTimeStampAsLong());
+        Vertex processInstance = addVertex(processInstanceName, RelationshipType.PROCESS_INSTANCE,
+                context.hasTimeStamp() ? context.getTimeStampAsLong() : null);
```

Comment:
This is the status update time, not the nominal time. We always have nominal time and status information, which are the most important.